codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/generator/library/ModelTestCodegen.java

```diff
@@ -29,9 +30,32 @@
  * e.g. generating JUnit tests from traits like @smithy.test#smokeTests.
  * This is distinct from the Dafny testing code
  * and the testing wrapper to support it generated by TestJavaLibrary.
+ *
+ * <h1>Smoke test generation</h1>
+ *
+ * If the {@code vendorParams} property is present,
+ * the generated test constructs the client config with its values.
+ *
+ * <h2>Assertions</h2>
+ *
+ * If {@code "INVALID_CONFIG"} is present in the test case's {@code tags},
+ * the generated test asserts that an exception is thrown
+ * during either client config construction or client construction,
+ * and the test neither constructs the operation input nor performs the operation call.
+ * <p>
+ * Otherwise, if the {@code expect.success} property is present,
+ * the generated test asserts that no exception is thrown
+ * during the operation call and all setup steps
+ * (construction of client config, client, and operation input).
+ * <p>
+ * Otherwise, if the {@code expect.failure} property is present,
+ * the generated test asserts that an exception is thrown
+ * during either operation input construction or the operation call.
```

Comment:
Clever solution!

The only issue is that `@smokeTests` is currently only implemented for Java (notice this file is under `.../smithyjava/...`, so this PR isn't actually testing the Rust support. And based on looking at the generated code I think you're missing the actual call to validate the input in `from_conf`.

I'd suggest adding a completely manual test or two to `Constraints/runtimes/rust/tests/simple_constraints_test.rs` to meet the testing bar, rather than taking on implementing `smokeTests` for Rust in this PR.

Reply:
Scratch that, for some reason when I code generated locally I didn't see the new validation right away. And having the smoke test support include this for Java is awesome, I'm not suggesting removing it.

Reply:
Thanks, and just to clarify, the manual Rust tests are already present: https://github.com/smithy-lang/smithy-dafny/pull/748/files#diff-d4d7fdb673501fddda845fbe959f8f559f69e314b8f77e77fb09293dbb49e81c

Reply:
Realized that too after the fact, was looking in the wrong place